leetcodePython/SlidingWindow/slidingwindow_2537.py

```diff
@@ -0,0 +1,19 @@
+import collections
+from typing import List
+from collections import Counter
+
+
+class Solution:
+    def countGood(self, nums: List[int], k: int) -> int:
+        ans = left = 0
+        count = Counter()
+        new_pairs = 0
+        for right in range(len(nums)):
+            new_pairs += count[nums[right]]
+            count[nums[right]] += 1
+            while new_pairs >= k:
+                count[nums[left]] -= 1
+                new_pairs -= count[nums[left]]
+                left += 1
+            ans += left
```

Comment:
The left index is when the condition is invalid for current `right`. So any index before current `left` is valid.